### PR TITLE
(slightly) Improving deb packaging with init scripts and user creation

### PR DESCRIPTION
**Pull request by mikn**
As mentioned in PR: #791 there was some wishes for upstart and systemd jobs as well as an init script and there was also concerns that the install([...]) used in the PR would actually have included the files in all packaging.
I've added in a CMake module that I found here: https://github.com/sebknzl/cmake-debhelper/ which is licensed under GPL (might be of concern? But it is only for the build system, so it should not go viral over the entire code base)
This module is "needed" to execute dh_* scripts during the setup of the project - those in turn are needed to be able to use the debian packaging idioms of putting the init scripts into a debian/ folder.

I also found a "bug" in the current deb-package setup. The listed debian dependency did not get assigned to the package, which I'll admit caused me some headache -  I initially tried adding the init files in a similar fashion. Passing it in as a variable to the custom make entry seems to have made it work, though.
But is it really needed to have libc6 above 2.13? It means that heka won't install on squeeze...

Though, in the long run I would really recommend going away from having CPack do the Debian packaging.
I didn't do that now since it would've involved rearranging a significant portion of the build, but if it is something you want to do with the project I think I could summon the time to do that too. :)

### CODE REVIEW

**Comment by nickchappell** [comment]
+1

**Comment by BartVB** [comment]
Would be great if this could be merged + new PR for upstart, most of the improvements have been ready/usable since october. It's a shame not to use them.

**Comment by rafrombrc** [comment]
Unfortunately, this isn't a priority for us, and the Heka team is too busy working on the core product to have any time left for focusing on the various different distributions. I'm happy to review and provide feedback, but for this to land it's going to need someone to step up and finish it up.

**Comment by davecap** [comment]
+1

**Comment by sathieu** [comment]
Why not simply drop the upstart part?

Upstart can deal with the init.d script, and upstart will not be in any upcoming big distribution (appart maybe from ChromeOS, ChromiumOS and WebOS).

File: debian/heka.service.in [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=heka - data collector and processor daemon
+After=network.target auditd.service
```
Comment (sathieu) [inline debian/heka.service.in]: Why auditd.service?

**Comment by rafrombrc** [comment]
Superseded by #1513.

**Pull request closed without merging**